src/Servers/Kestrel/Core/src/KestrelServerOptions.cs

```diff
@@ -65,11 +74,28 @@ public class KestrelServerOptions
         /// </remarks>
         public bool DisableStringReuse { get; set; } = false;
 
+        /// <summary>
+        /// Controls whether to return the AltSvcHeader from on an HTTP/2 or lower response for HTTP/3
+        /// </summary>
+        /// <remarks>
+        /// Defaults to false.
+        /// </remarks>
+        public bool EnableAltSvc { get; set; } = false;
+
+        /// <summary>
+        /// Gets or sets a callback that returns the <see cref="Encoding"/> to decode the value for the specified request header name,
+        /// or <see langword="null"/> to use the default <see cref="UTF8Encoding"/>.
+        /// </summary>
+        /// <remarks>
+        /// Defaults to returning a <see langword="null"/> for all headers.
+        /// </remarks>
```

Comment:
You already said above that UTF-8 is the default. Besides, DefaultRequestHeaderEncodingSelector doesn't return null, it returns UTF8EncodingSealed.
```suggestion
```

Reply:
Locally, I do have DefaultRequestHeaderEncodingSelector returning null for all headers, and I plan to push that, but I can remove the remark.